flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MultipleProgramsTestBaseJUnit5.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.test.util;
+
+import org.apache.flink.testutils.junit.extensions.parameterized.Parameter;
+import org.apache.flink.testutils.junit.extensions.parameterized.Parameters;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ * Base class for unit tests that run multiple tests and want to reuse the same Flink cluster. This
+ * saves a significant amount of time, since the startup and shutdown of the Flink clusters
+ * (including actor systems, etc) usually dominates the execution of the actual tests.
+ *
+ * <p>To write a unit test against this test base, simply extend it and add one or more regular test
+ * methods and retrieve the ExecutionEnvironment from the context:
+ *
+ * <pre>{@code
+ * {@literal @}Test
+ * public void someTest() {
+ *     ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+ *     // test code
+ *     env.execute();
+ * }
+ *
+ * {@literal @}Test
+ * public void anotherTest() {
+ *     ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+ *     // test code
+ *     env.execute();
+ * }
+ *
+ * }</pre>
+ */
+public class MultipleProgramsTestBaseJUnit5 extends AbstractTestBaseJUnit5 {
+
+    /**
+     * Enum that defines which execution environment to run the next test on: An embedded local
+     * flink cluster, or the collection execution backend.
+     */
+    public enum TestExecutionMode {
```

Comment:
Can this enum be pulled out from `MultipleProgramsTestBaseJUnit5` ?
 Might be useful for future extension or other new test cases for future.

Reply:
If I understand correctly, the enum should be the logic inside of `MultipleProgramsTestBase` or its inheritor.

So leaving it here looks good to me.